### PR TITLE
Remove findDOMNode usage from the inserter

### PR DESCRIPTION
**Pull request by youknowriad**
Related #11360 

This PR leverages `forwardRef` to remove `findDOMNode` usage from the inserter. The idea is that you can `ref` a `PanelBody` component and you get the inner div instead. It's a small breaking change but since we never documented `ref` as a public prop for `PanelBody` I don't expect this to break anything.

### CODE REVIEW

**Review by jorgefilipecosta — COMMENTED** [review]
Hi @youknowriad in my tests this is working correctly 👍
Normally we use createRef() to initilize the references would this apporach be equivalent:
image

File: packages/editor/src/components/panel-color-settings/test/__snapshots__/index.js.snap [hunk 1]
```diff
@@ -87,7 +87,7 @@ exports[`PanelColorSettings should render a color panel if at least one setting
 `;
 
 exports[`PanelColorSettings should render a color panel if at least one setting specifies some colors to choose 2`] = `
-<PanelBody
+<[object Object]
```
Comment (jorgefilipecosta) [inline packages/editor/src/components/panel-color-settings/test/__snapshots__/index.js.snap]: I wonder why the displayName we set is not being used here.

Comment (youknowriad) [reply packages/editor/src/components/panel-color-settings/test/__snapshots__/index.js.snap]: Probably an enzyme bug :)

**Comment by youknowriad** [comment]
@jorgefilipecosta I don't know if it's equivalent. In general the `createRef` is intialized in the constructor while in this case we don't know all the instances we want to create.

**Comment by jorgefilipecosta** [comment]
> @jorgefilipecosta I don't know if it's equivalent. In general the createRef is intialized in the constructor while in this case we don't know all the instances we want to create.

That's true, nothing says it must be initialized in the constructor but your approach seems safer.

**Review by jorgefilipecosta — APPROVED** [review]
LGTM 👍

**Pull request merged**